File: tests/fixtures/analyzers/go/decisions.go
package sample

func DeeplyNested(values []int) int {
	if len(values) > 0 {
		for _, value := range values {
			for value > 0 {
				if value%2 == 1 {
					return value
				}
				value--
			}
		}
	}
	return 0
}

func Branching(a, b, c, d bool) int {
	result := 0
	if a { result++ }
	if b { result++ }
	if c { result++ }
	if d { result++ }
	return result
}

func Expressions(a, b, c bool) bool {
	return a && b || c
}

func Switches(value any) int {
	switch typed := value.(type) {
	case int:
		return typed
	case string:
		return len(typed)
	default:
		return 0
	}
}

File: tests/fixtures/analyzers/python/decisions.py
def deeply_nested(value):
    if value:
        for item in value:
            while item > 0:
                if item % 2:
                    return item
                item -= 1
    return 0


def branching(a, b, c, d):
    if a:
        a = 0
    elif b:
        b = 0
    if c:
        c = 0
    if d:
        d = 0
    return a + b + c + d


def expressions(items, left, right):
    selected = [item for item in items if item > 0]
    choice = left if left and right else right
    return selected, choice


def patterns(value):
    try:
        match value:
            case 1:
                return "one"
            case 2 | 3:
                return "few"
            case _:
                return "other"
    except ValueError:
        return "invalid"
